chemprop/train/loss_functions.py

```diff
@@ -171,16 +223,217 @@ def wasserstein_loss(model_spectra: torch.tensor, target_spectra: torch.tensor,
+    :lambda: coefficient to weight KL term
+
+    :return: Loss
+    """
+
+    def KL(alpha):
+        """
+        Compute KL for Dirichlet defined by alpha to uniform dirichlet
+        :alpha: parameters for Dirichlet
+
+        :return: KL
+        """
+        beta = torch.ones_like(alpha)
+        S_alpha = torch.sum(alpha, dim=-1, keepdim=True)
+        S_beta = torch.sum(beta, dim=-1, keepdim=True)
+
+        ln_alpha = torch.lgamma(S_alpha) - torch.sum(
+            torch.lgamma(alpha), dim=-1, keepdim=True
+        )
+        ln_beta = torch.sum(torch.lgamma(beta), dim=-1, keepdim=True) - torch.lgamma(
+            S_beta
+        )
+
+        # digamma terms
+        dg_alpha = torch.digamma(alpha)
+        dg_S_alpha = torch.digamma(S_alpha)
+
+        # KL
+        kl = (
+            ln_alpha
+            + ln_beta
+            + torch.sum((alpha - beta) * (dg_alpha - dg_S_alpha), dim=-1, keepdim=True)
+        )
+        return kl
+
+    num_tasks = target_labels.shape[1]
+    num_classes = 2
+    alphas = torch.reshape(alphas, (alphas.shape[0], num_tasks, num_classes))
+
+    y_one_hot = torch.eye(num_classes)[target_labels.long()]
+    if target_labels.is_cuda:
+        y_one_hot = y_one_hot.cuda()
+
+    # SOS term
+    S = torch.sum(alphas, dim=-1, keepdim=True)
+    p = alphas / S
+    A = torch.sum(torch.pow((y_one_hot - p), 2), dim=-1, keepdim=True)
+    B = torch.sum((p * (1 - p)) / (S + 1), dim=-1, keepdim=True)
+    SOS = A + B
+
+    # KL
+    alpha_hat = y_one_hot + (1 - y_one_hot) * alphas
+    KL = lam * KL(alpha_hat)
+
+    # loss = torch.mean(SOS + KL)
+    loss = SOS + KL
+    loss = torch.mean(loss, dim=-1)
+    return loss
+
+
+def dirichlet_multiclass_loss(alphas, target_labels, lam=0):
+    """
+    Use Evidential Learning Dirichlet loss from Sensoy et al for multiclass datasets.
+    :param alphas: Predicted parameters for Dirichlet in shape(datapoints, task, classes).
+    :param target_labels: Digital labels to predict in shape(datapoints, tasks).
+    :lambda: coefficient to weight KL term
+
+    :return: Loss
+    """
+
+    def KL(alpha):
+        """
+        Compute KL for Dirichlet defined by alpha to uniform dirichlet
+        :alpha: parameters for Dirichlet
+
+        :return: KL
+        """
+        beta = torch.ones_like(alpha)
+        S_alpha = torch.sum(alpha, dim=-1, keepdim=True)
+        S_beta = torch.sum(beta, dim=-1, keepdim=True)
+
+        ln_alpha = torch.lgamma(S_alpha) - torch.sum(
+            torch.lgamma(alpha), dim=-1, keepdim=True
+        )
+        ln_beta = torch.sum(torch.lgamma(beta), dim=-1, keepdim=True) - torch.lgamma(
+            S_beta
+        )
+
+        # digamma terms
+        dg_alpha = torch.digamma(alpha)
+        dg_S_alpha = torch.digamma(S_alpha)
+
+        # KL
+        kl = (
+            ln_alpha
+            + ln_beta
+            + torch.sum((alpha - beta) * (dg_alpha - dg_S_alpha), dim=-1, keepdim=True)
+        )
+        return kl
```

Comment:
looks like you don't use it once, but just redefine it in every function that uses it?